### PR TITLE
[13.0] EDI add file reception + auto generate ack files when needed

### PR DESCRIPTION
**Pull request by simahawk**
Based on/includes #271 (to be merged 1st).
Only last 4 commits are relevant (excluding a fixup)

### CODE REVIEW

File: edi/models/edi_backend.py [hunk 1]
```diff
@@ -358,3 +364,122 @@ def _exchange_process(self, exchange_record):
...
+        component = self._get_component(exchange_record, "receive")
+        if component:
+            return component.receive()
+        raise NotImplementedError()
+
+    def _cron_check_input_exchange_sync(self, **kw):
+        for backend in self:
+            backend._check_input_exchange_sync(**kw)
+
+    def _check_input_exchange_sync(self):
```
Comment (etobella) [inline edi/models/edi_backend.py]: Missing the **kw, as they might be passed by the cron?

File: edi/models/edi_backend.py [hunk 2]
```diff
@@ -358,3 +364,122 @@ def _exchange_process(self, exchange_record):
...
+        ]
+
+    def _exchange_receive(self, exchange_record):
+        component = self._get_component(exchange_record, "receive")
+        if component:
+            return component.receive()
+        raise NotImplementedError()
+
+    def _cron_check_input_exchange_sync(self, **kw):
+        for backend in self:
```
Comment (etobella) [inline edi/models/edi_backend.py:425]: I think related cron is commented, with this implementation we might add it

**Review by etobella — APPROVED** [review]
Just some minor comments

File: edi/models/edi_backend.py (hunk 2 above)
Comment (simahawk) [reply edi/models/edi_backend.py:425]: I prefer to wait when tests will be there :wink:

File: edi/models/edi_backend.py (hunk 1 above)
Comment (simahawk) [reply edi/models/edi_backend.py]: good catch... you see? No tests... will come soon on next round :wink:

**Pull request merged**